api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -33,6 +33,16 @@
  */
 public interface Context
 {
+  /*
```

Comment:
I'm not sure I completely understand Point 3. I provide a solution to point 2.

The properties file will now have an option to specify a fully qualified attribute name. A fully qualified attribute name is an attribute name prefixed by the class name. For example:

com.datatorrent.api.Context.OperatorContext.AUTO_RECORD specifies the Operator AUTO_RECORD attribute.

com.datatorrent.api.Context.PortContext.AUTO_RECORD specifies the Port AUTO_RECORD attribute.

So if a properties file specified the following: dt.attr.com.datatorrent.api.Context.OperatorContext.AUTO_RECORD Then the AUTO_RECORD attribute will be set for operators.

If a properties file specifies this: dt.attr.AUTO_RECORD then the AUTO_RECORD attribute will be set for both ports and operators.

If we don't require Attributes with the same name to have the same type then we would have to be careful to not add a new Attribute with the same name but different type in a different context.